components/provisioner/internal/gardener/provisioner.go

```diff
@@ -63,6 +63,9 @@ func (g *GardenerProvisioner) ProvisionCluster(cluster model.Cluster, operationI
 
 	annotate(shootTemplate, operationIdAnnotation, operationId)
 	annotate(shootTemplate, runtimeIdAnnotation, cluster.ID)
+	if cluster.LicenceType != "" {
```

Comment:
We could simplify this design a bit. 
Some facts: 
- LicenceType is a field that is specific to GardenerConfig
- LicenseType annotation could be added in ToShootTemplate
- Annotations for OperationID and RuntimeID are added in ProvisionCluster method since those values are not available in ToShootTemplate

My proposal is to add an annotation in ToShootTemplate implementation. It will have the following implications:
- No need to change model.Cluster 
- No need to add GetLicenceType method on ProviderConfiguration
- No need to change gardener.ProvisionCluster method.